### PR TITLE
Change fp16 `atomic_rmw` error to warning

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/PipelineManager.h [hunk 1]
```diff
@@ -99,6 +99,9 @@ struct FuncOpConversion : public ConvertOpToLLVMPattern<triton::FuncOp> {
     if (!LLVM::isKernel(funcOp))
       amendedFuncOp = amendFuncOp(funcOp, rewriter);
 
+    for (unsigned i = 0; i < funcOp.getNumArguments(); ++i)
+      amendedFuncOp.removeArgAttr(i, "tt.divisibility");
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/PipelineManager.h:103]: remove `tt.divisibility` attributes to avoid unhandled warnings.

**Review by etiotto — APPROVED** [review]

File: python/src/ir.cc [hunk 2]
```diff
@@ -1573,11 +1573,11 @@ void init_triton_ir(py::module &&m) {
...
                context->printStackTraceOnDiagnostic(true);
-               context->getDiagEngine().registerHandler([](Diagnostic &diag) {
-                 llvm::outs() << diag << "\n";
-                 return success();
-               });
              }
+             context->getDiagEngine().registerHandler([](Diagnostic &diag) {
+               llvm::outs() << diag << "\n";
+               return success();
+             });
```
Comment (anmyachev) [inline python/src/ir.cc:1580]: @whitneywhtsang could we revert this change?

Comment (whitneywhtsang) [reply python/src/ir.cc:1580]: if we revert this change, then the warning won't be shown unless user uses an env var. 
@etiotto What's your opinion?

Comment (etiotto) [reply python/src/ir.cc:1580]: For the sake of removing diffs with upstream code we can put it in the if statement. The warning will come out only is `haveDiagnostics` is set.

Comment (whitneywhtsang) [reply python/src/ir.cc:1580]: Created https://github.com/intel/intel-xpu-backend-for-triton/pull/2943.
